Feature/hw 04

uoles opened the pull request.

uoles committed:
Add files commit.

uoles committed:
Edit readme.

uoles committed:
Edit readme.

uoles committed:
Edit readme.

uoles committed:
- Переименование классов.
- Добавление исключения QuestionsFileLoadingException.
- Работа с консолью вынесена в сервис ConsoleService.
- Тесты для QuestionsDAO.

Commit:
- Переименованы классы.
- Добавлены тесты.

Commit:
- XML конфигурация spring заменена на Java-based.
- Изменения в структуре файлов.

Commit:
Java-based конфигурация spring заменена на Annotation-based.

uoles committed:
Загрузка имени файла с вопросами из конфигурационного файла.

Commit:
- Рефакторинг.
- Использование библиотеки lombok.

Commit:
- Локализация для вопросов и сообщений.
- Файл настроек.
- Java + Annotation-based конфигурация.

uoles committed:
Merge branch 'master' into feature/hw-02

uoles committed:
- Убраны лишние тесты.
- Изменена работа с файлом.
- Использование @Data вместо @Getter, @Setter.

uoles committed:
Рефакторинг.

uoles committed:
Merge branch 'master' into feature/hw-02

Commit:
Добавление файлов настроек для тестов.

Commit:
Рефакторинг.

uoles committed:
Рефакторинг.

uoles committed:
- Перенос приложения на Spring Boot.
- Добавление своего баннера.

Commit:
Тесты переведены на Spring Boot.

Commit:
Настройки перенесены в *.yml файлы.

Commit:
Добавлен баннер.

Commit:
Рефакторинг.

Commit:
Доработки по замечаниям.

uoles committed:
Добавление Spring Shell.

uoles committed:
Решение проблемы блокирования тестов Spring Shell'ом.

stvort commented on src/main/java/ru/otus/mkulikov/app/services/questions/QuestionsServiceImpl.java:
```diff
@@ -0,0 +1,74 @@
+            consoleService.write(c_delimeter);
+        }
+
+        long count = questions.stream()
```
Функционал по печати результатов лучше вынести в отдельный метод

stvort commented on src/main/java/ru/otus/mkulikov/shell/Commands.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @ShellMethod(key = { "startTest" }, value = "Start test")
+    public void startTest() {
```
Если разделить  ```processor.startTest();``` Хотя бы на регистрацию и начало теста, то появится замечательная возможность попробовать обеспечить очередность команд с помощью ```@ShellMethodAvailability```

stvort commented on src/test/java/ru/otus/mkulikov/app/services/AppTestConfig.java:
```diff
@@ -0,0 +1,20 @@
+@SpringBootConfiguration
+@ComponentScan(basePackages = "ru.otus.mkulikov.app")
+@EnableConfigurationProperties
+public class AppTestConfig {
```
Кстати команды шелл тоже можно тестировать. Для этого можно внедрить в тест ```@Autowired Shell shell;``` и далее ```shell.evaluate(() -> "startTest");```

uoles committed:
- Функционал по печати результатов вынесен в отдельный метод.
- Разделение функционала метода startTest и использование @ShellMethodAvailability.
